packages/apollo-gateway/src/__tests__/integration/networkRequests.test.ts

```diff
@@ -0,0 +1,95 @@
+import nock from 'nock';
+import { ApolloGateway } from '../..';
+import createSHA from '../../utilities/createSHA';
+
+import {
+  mockGetRawPartialSchema,
+  mockFetchStorageSecret,
+  mockGetCompositionConfigLink,
+  mockGetCompositionConfigs,
+  mockGetImplementingServices,
+  mockLocalhostSDLQuery,
+} from './nockMocks';
+
+it('Queries remote endpoints for their SDLs', async () => {
+  let url = 'http://localhost:4001';
+  let sdl = `
+  extend type Query {
+      me: User
+      everyone: [User]
+  }
+
+  "My User."
+  type User @key(fields: "id") {
+    id: ID!
+    name: String
+    username: String
+  }
+  `;
+
+  mockLocalhostSDLQuery({ url, sdl });
+
+  let gateway = new ApolloGateway({
+    serviceList: [{ name: 'accounts', url: `${url}/graphql` }],
+  });
+  await gateway.load();
+  expect(nock.isDone()).toBeTruthy();
+  expect(gateway.schema!.getType('User')!.description).toBe('My User.');
+});
+
+// This test is maybe a bit terrible, but IDK a better way to mock all the requests
+it('Extracts service definitions from remote storage', async () => {
+  let serviceName = 'jacksons-service';
+  let apiKey = `service:${serviceName}:AABBCCDDEEFFGG`;
+  let apiKeyHash = createSHA('sha512')
+    .update(apiKey)
+    .digest('hex');
+
+  let storageSecret = 'secret';
+  let implementingServicePath = 'path-to-implementing-service-definition.json';
+  let partialSchemaPath = 'path-to-accounts-partial-schema.json';
+  let federatedServiceName = 'accounts';
+  let federatedServiceURL = 'http://localhost:4001';
```

Comment:
Some of the multi-line declarations make sense, and those that are used more than once, but most of these separate variable declarations don't seem to contribute much value?  Seems they could be `const`ants where appropriate too, which provides immediate confirmation to the reader that they won't be manipulated.

Reply:
Made them all `const`. The `federatedServiceSchemaURL` and `federatedServiceSchema` are the only ones used once... I'd prefer to define all the options together rather than have almost all of them defined at top, but then a couple passed directly.